os/drivers/video/video.c

```diff
@@ -326,11 +329,8 @@ static int change_video_state(FAR video_upperhalf_t *priv, enum video_state_e ne
 		}
 	}
 
-	if (ret == OK) {
-		priv->video_inf.state = updated_next_state;
-	}
-
-	return ret;
+	priv->video_inf.state = updated_next_state;
```

Comment:
it is state changing, lock mechanism should be there.

Reply:
change_video_state function is called always by caller with adequate locking, hence no need to do inside the function.